python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -135,19 +134,14 @@ cdef class GroupBy:
                 else _DECIMAL_AGGS if is_decimal_dtype(dtype)
                 else "ALL"
             )
-            if (valid_aggregations is _DECIMAL_AGGS
-                    and rmm._cuda.gpu.runtimeGetVersion() < 11000):
-                raise RuntimeError(
-                    "Decimal aggregations are only supported on CUDA >= 11 "
-                    "due to an nvcc compiler bug."
-                )
 
             c_agg_request = move(libcudf_groupby.aggregation_request())
+            included_aggregations.append([])
             for agg in aggs:
                 agg_obj = make_groupby_aggregation(agg)
                 if (valid_aggregations == "ALL"
                         or agg_obj.kind in valid_aggregations):
-                    included_aggregations[col_name].append(agg)
+                    included_aggregations[-1].append(agg)
```

Comment:
Instead of appending an empty list of aggregations to `included_aggregations` before the loop and then indexing into it, store the empty list in a temp variable after `for agg in aggs` and then append it to `included_aggregations` at the end of the loop.